pkg/handlers/helper.go

```diff
@@ -0,0 +1,224 @@
+	SendServiceLog(n *oav1alpha1.Notification, firing bool) error
+}
+
+type ocmsdkclient struct {
+	ocm *sdk.Connection
+}
+
+type WebhookReceiverHandler struct {
+	c   client.Client
+	ocm OCMClient
+}
+
+type OCMResponseBody struct {
+	Reason string `json:"reason"`
+}
+
+func NewOcmClient(ocm *sdk.Connection) OCMClient {
+	return &ocmsdkclient{
+		ocm: ocm,
+	}
+}
+
+// isValidAlert indicates whether the supplied alert is one that warrants being processed for a notification.
+// Any or all of these situations should be treated as an error as it indicates that AlertManager is forwarding
+// alerts to ocm-agent that it should not be.
+func isValidAlert(alert template.Alert) bool {
+	// An invalid alert won't have a name
+	alertname, err := alertName(alert)
+	if err != nil {
+		log.WithError(err).Info("alertname missing for alert")
+		return false
+	}
+
+	// An invalid alert won't have a send_managed_notification label
+	if val, ok := alert.Labels[AMLabelManagedNotification]; !ok || val == "false" {
+		log.WithField(LogFieldAlertname, alertname).Error("alert has no send_managed_notification label")
+		return false
+	}
+
+	// An invalid alert won't have a managed_notification_template label
+	if _, ok := alert.Labels[AMLabelTemplateName]; !ok {
+		log.WithField(LogFieldAlertname, alertname).Error("alert has no managed notification defined")
+		return false
+	}
+
+	return true
+}
+
+// alertName looks up the name of an AlertManager alert, or returns error if one does not exist
+func alertName(a template.Alert) (*string, error) {
+	if name, ok := a.Labels[AMLabelAlertName]; ok {
+		return &name, nil
+	}
+	return nil, fmt.Errorf("no alertname defined in alert")
+}
+
+// SendServiceLog sends a servicelog notification for the given alert
+func (o *ocmsdkclient) SendServiceLog(n *oav1alpha1.Notification, firing bool) error {
+	req := o.ocm.Post()
+	err := arguments.ApplyPathArg(req, "/api/service_logs/v1/cluster_logs")
+	if err != nil {
+		return err
+	}
+
+	sl := ocm.ServiceLog{
+		ServiceName:  consts.ServiceLogServiceName,
+		ClusterUUID:  viper.GetString(config.ClusterID),
+		Summary:      n.Summary,
+		InternalOnly: false,
+	}
+
+	// Use different Summary and Description for firing and resolved status for an alert
+	if firing {
+		sl.Description = n.ActiveDesc
+		sl.Summary = ServiceLogActivePrefix + ": " + n.Summary
+	} else {
+		sl.Description = n.ResolvedDesc
+		sl.Summary = ServiceLogResolvePrefix + ": " + n.Summary
+	}
+	slAsBytes, err := json.Marshal(sl)
+	if err != nil {
+		return err
+	}
+
+	req.Bytes(slAsBytes)
+
+	res, err := req.Send()
+	if err != nil {
+		return err
+	}
+
+	operationId := res.Header(HeaderOperationId)
+	err = responseChecker(operationId, res.Status(), res.Bytes())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SendServiceLog sends a servicelog notification for the given alert in fleetmode
+func (o *ocmsdkclient) SendServiceLogInFleetMode(n *oav1alpha1.FleetNotification, firing bool, clusterID string) error {
```

Comment:
We can maybe avoid some code duplication here by unifying the two `SendServiceLog..` functions.

- Introduce/keep the `clusterID` parameter, like this function does, but change the non-RHOBS handler to supply the clusterID to it.
- The `if not firing, then generate resolved SL` logic can remain in there rather than be removed, because the RHOBS handler will only ever call the function when an alert is firing (and if it calls the handler with `firing` set to false then that's a bug in the RHOBS handler)

Reply:
Ah, now I see why this was done. The different in types for the Notification passed through. :)  
I'll have a think about if we can still improve here.

Reply:
(this is now resolved, but feedback on the chosen solution of changing the function signature is welcome)